### PR TITLE
OSF-5505 GDrive "def empty_folder" for use in "def delete" against root

### PR DESCRIPTION
**Pull request by TomBaxter**
## Purpose
[OSF-5505](https://openscience.atlassian.net/browse/OSF-5505)
Catch delete calls against provider root in GoogleDrive to prevent from actually deleting the provider root. Just delete contents of provider root

## Changes
Add def empty_folder for use by def delete in the case of root provider

## Side effects
None

#OSF-5505

### CODE REVIEW

File: waterbutler/providers/googledrive/provider.py [hunk 1]
```diff
@@ -206,9 +206,25 @@ def upload(self, stream, path, **kwargs):
...
+            If the WaterButlerPath given is for the provider root path, then
+            the contents of provider root path will be deleted. But not the
+            provider root itself.
+        """
         if not path.identifier:
             raise exceptions.NotFoundError(str(path))
 
+        if path.is_root:
+            yield from self.empty_folder(path)
+            return
```
Comment (felliott) [inline waterbutler/providers/googledrive/provider.py]: This should return a 204 response

File: waterbutler/providers/googledrive/provider.py [hunk 2]
```diff
@@ -216,6 +232,39 @@ def delete(self, path, **kwargs):
...
+    def empty_folder(self, path):
+        """Given a WaterButlerPath, delete all contents of folder
+
+        :param WaterButlerPath: Folder to be emptied
+        :rtype: None
+        :raises: :class:`waterbutler.core.exceptions.NotFoundError`
+        :raises: :class:`waterbutler.core.exceptions.MetadataError`
+        :raises: :class:`waterbutler.core.exceptions.DeleteError`
+        """
+        fid = path.identifier
```
Comment (felliott) [inline waterbutler/providers/googledrive/provider.py]: Please rename `fid` to something more informative like `file_id`.

File: waterbutler/providers/googledrive/provider.py (hunk 1 above)
Comment (TomBaxter) [inline waterbutler/providers/googledrive/provider.py]: def empty_folder returns 204. I tested on my laptop and waterbutler logs show 204 response. If I'm missing something here, I'm not sure what.

File: waterbutler/providers/googledrive/provider.py (hunk 2 above)
Comment (TomBaxter) [inline waterbutler/providers/googledrive/provider.py]: Done

**Comment by felliott** [comment]
The scope of the original issue has changed.  This PR is being closed and a new PR will be opened later including this and the other delete-storage-root PRs.

**Pull request closed without merging**